### ISSUE TITLE
Followed exactly and the app crashes

### ISSUE THREAD

**Issue by rdholland** [MicrosoftDocs/windows-uwp#2733]
Access violation. It doesn't get much more basic than the outlined steps.

template <typename D> WINRT_IMPL_AUTO(void) consume_Microsoft_UI_Xaml_IApplicationStatics<D>::Start(Microsoft::UI::Xaml::ApplicationInitializationCallback const& callback) const
    {
        **check_hresult(WINRT_IMPL_SHIM(Microsoft::UI::Xaml::IApplicationStatics)->Start(*(void**)(&callback)));**
    }

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8c6a5c75-6cdc-a97f-df50-938ac5534e28
* Version Independent ID: e7195fca-7687-4296-6aa8-638ce198d549
* Content: [Get started with WinUI 3 for desktop apps](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/get-started-winui3-for-desktop)
* Content Source: [hub/apps/winui/winui3/get-started-winui3-for-desktop.md](https://github.com/MicrosoftDocs/windows-uwp/blob/docs/hub/apps/winui/winui3/get-started-winui3-for-desktop.md)
* Product: **uwp**
* Technology: **apps**
* GitHub Login: @mcleanbyron
* Microsoft Alias: **mcleans**

**Comment by mcleanbyron** [comment]
I'm sorry you're running into this issue. Based on the code you referenced in your post, it sounds like you're encountering this error when trying to build and run a "Blank App, Packaged (WinUI in Desktop)" project using C++. Is that correct? 

I just tested this again and I was able to build and run the project without errors. Can you confirm that you followed all of the [setup instructions](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/#install-winui-3-preview-2), and the versions of Visual Studio 2019 and the **WinUI 3 Project Templates [Preview]** extension you have installed?

**Comment by rdholland** [comment]
I have VS 16.7.4 and again followed the steps found at https://docs.microsoft.com/en-us/windows/apps/winui/winui3/get-started-winui3-for-desktop

I made sure I had the .NET Preview 5 x86 and x64 packages. I tried again and got the exact same results.

Unhandled exception at 0x00007FF95D2F6589 (Microsoft.Internal.XamlUdk.dll) in WinUITake2.exe: 0xC000027B: An application-internal exception has occurred (parameters: 0x0000018ECB81D360, 0x0000000000000001).

I searched for that error and followed a number of the links but didn't see anything that looked like my crash in the template <typename D> WINRT_IMPL_AUTO(void) consume_Microsoft_UI_Xaml_IApplicationStatics<D>::Start(Microsoft::UI::Xaml::ApplicationInitializationCallback const& callback) const

Immediate call stack is:

 	KernelBase.dll!RaiseFailFastException()	Unknown
 	combase.dll!RoFailFastWithErrorContextInternal2(HRESULT hrError, unsigned long cStowedExceptions=0, _STOWED_EXCEPTION_INFORMATION_V2 * * aStowedExceptionPointers) Line 1458	C++
 	Microsoft.Internal.XamlUdk.dll!FailFastWithStowedExceptionInfo()	Unknown
 	Microsoft.UI.Xaml.dll!UdkShim::FailFastWithStowedExceptions(HRESULT hrError=REGDB_E_CLASSNOTREG Class not registered, unsigned long cStowedExceptions=1, _STOWED_EXCEPTION_INFORMATION_V2 * * aStowedExceptionPointers=0x0000018ecb81d360) Line 85	C++
 	Microsoft.UI.Xaml.dll!FailFastWithStowedExceptions(HRESULT hrError=REGDB_E_CLASSNOTREG Class not registered, unsigned long cStowedExceptions=1, _STOWED_EXCEPTION_INFORMATION_V2 * * aStowedExceptionPointers=0x0000018ecb81d360) Line 93	C++
 	Microsoft.UI.Xaml.dll!CXcpDispatcher::Init(IXcpHostSite * pSite) Line 356	C++
 	Microsoft.UI.Xaml.dll!CXcpDispatcher::Create(IXcpHostSite * pSite=0x0000018ecb7be458, IXcpDispatcher * * ppInterface=0x0000018ecb7be480) Line 505	C++
 	Microsoft.UI.Xaml.dll!CJupiterControl::Init() Line 145	C++
 	Microsoft.UI.Xaml.dll!CJupiterControl::Create(CJupiterControl * * ppControl=0x0000018ecb7eb668) Line 32	C++
 	Microsoft.UI.Xaml.dll!DirectUI::DXamlCore::InitializeInstance(InitializationType initializationType=IslandsOnly, bool runWithoutCoreApplication=true) Line 372	C++
 	Microsoft.UI.Xaml.dll!DirectUI::DXamlCore::InitializeImpl(InitializationType initializationType=IslandsOnly, bool runWithoutCoreApplication=true) Line 2636	C++
 	Microsoft.UI.Xaml.dll!DirectUI::FrameworkApplication::StartDesktop() Line 188	C++
 	Microsoft.UI.Xaml.dll!DirectUI::FrameworkApplicationFactory::Start(ABI::Microsoft::UI::Xaml::IApplicationInitializationCallback * pCallback=0x0000018ecb7c75b0) Line 713	C++
>	WinUITake2.exe!winrt::impl::consume_Microsoft_UI_Xaml_IApplicationStatics<winrt::Microsoft::UI::Xaml::IApplicationStatics>::Start(const winrt::Microsoft::UI::Xaml::ApplicationInitializationCallback & callback={...}) Line 260	C++

Perhaps version info on the DLLs involved helps.

Microsoft.Internal.XamlUdk.dll	Microsoft.Internal.XamlUdk.dll	C:\Users\rdhollan\source\repos\WinUITake2\x64\Debug\WinUITake2\Microsoft.Internal.XamlUdk.dll	N/A	Yes	Symbols loaded.	C:\Users\rdhollan\AppData\Local\Temp\SymbolCache\Microsoft.Internal.XamlUDK.pdb\6916872204B652513B2AAF18996A8AA71\Microsoft.Internal.XamlUDK.pdb	46		<Unknown>	00007FF95D2F0000-00007FF95D303000	[11680] WinUITake2.exe		
Microsoft.UI.Xaml.dll	Microsoft.UI.Xaml.dll	C:\Users\rdhollan\source\repos\WinUITake2\x64\Debug\WinUITake2\Microsoft.UI.Xaml.dll	N/A	Yes	Symbols loaded.	C:\Users\rdhollan\AppData\Local\Temp\SymbolCache\Microsoft.ui.xaml.pdb\B8441781FEFFA75A2C2E26D923C247321\Microsoft.ui.xaml.pdb	25	3.0.0-preview2.2007 (DCPP(PKGESNAT006)(issblder).160101.0800)	<Unknown>	00007FF908110000-00007FF908F30000	[11680] WinUITake2.exe		
Microsoft.UI.Xaml.Internal.dll	Microsoft.UI.Xaml.Internal.dll	C:\Users\rdhollan\source\repos\WinUITake2\x64\Debug\WinUITake2\Microsoft.UI.Xaml.Internal.dll	N/A	Yes	Cannot find or open the PDB file.		38		6/26/2020 12:54 PM	00007FF936E70000-00007FF936F8C000	[11680] WinUITake2.exe

**Comment by mcleanbyron** [comment]
Thanks for the additional info. Unfortunately, I can't reproduce the error your seeing. Because this repo is specifically intended for documentation feedback and not technical support issues, your best option is to report this error in a new issue in the [GitHub repo for WinUI](https://github.com/microsoft/microsoft-ui-xaml/issues). The engineering team that owns WinUI (including the new WinUI 3 project system) maintains that repo and investigates/responds to customer issues there.

**Comment by rdholland** [comment]
I finally got it to work. I must have missed a step in the turorial that said I have to enable "developer mode". There was a mysterious "Settings dialog" I noticed on one of my monitors when I opened the project again. It was related to the "(Package)" project the wizard tossed into the solution file. I thought it was just like the older (now disappeared) "Setup" projects one used to be able to insert into a solution to build a setup exe for a project. Turns out it isn't something you do after you create, build and test. It has to be executed first. In any case, I saw the window again and turned on developer mode. When I closed the dialog I still got an error in the deployment output window. But, after a couple of seconds, output started streaming to the output (Deployment) window and the application then started and ran without crashing.

So, my box is configured correctly and I can run a WinUI3 app now that I have "developer mode" on. If that step is in the tutorial, I missed it totally. If it is not, perhaps it should be added.
